packages/components/src/custom-gradient-picker/gradient-bar/control-points.tsx

```diff
@@ -148,142 +172,158 @@ function ControlPoints( {
 	// Adding `cleanEventListeners` to the dependency array below requires the function itself to be wrapped in a `useCallback`
 	// This memoization would prevent the event listeners from being properly cleaned.
 	// Instead, we'll pass a ref to the function in our `useEffect` so `cleanEventListeners` itself is no longer a dependency.
-	const cleanEventListenersRef = useRef();
+	const cleanEventListenersRef = useRef< ( () => void ) | undefined >();
```

Comment:
Small nit: the `| undefined` type is redundant, TypeScript would be able to infer that on its own

```suggestion
	const cleanEventListenersRef = useRef< () => void >();
```